ui/app/AppLayouts/Browser/JSDialogWindow.qml

```diff
@@ -0,0 +1,83 @@
+import QtQuick 2.13
+import QtQuick.Controls 2.13
+import QtQuick.Layouts 1.13
+import QtWebEngine 1.10
+import "../../../shared"
+import "../../../imports"
+
+ModalPopup {
+    id: root
+    property QtObject request
+    height: 286
+
+    closePolicy: Popup.NoAutoClose
+
+    onClosed: {
+        request.dialogReject();
+        root.destroy();
+    }
+
+    Component.onCompleted: {
+        root.title = request.securityOrigin;
+        message.text = request.message;
+        if(request.type === JavaScriptDialogRequest.DialogTypeAlert){
+            cancelButton.visible = false;
+        }
+        if(request.type === JavaScriptDialogRequest.DialogTypePrompt){
+            prompt.text = request.defaultText;
```

Comment:
I think a combination of request.securityOrigin and request.type would help users to know who they allow what with the prompt